ibis/backends/oracle/compiler.py

```diff
@@ -0,0 +1,617 @@
+from __future__ import annotations
+
+from functools import singledispatchmethod
+
+import sqlglot as sg
+import sqlglot.expressions as sge
+import toolz
+from public import public
+from sqlglot.dialects import Oracle
+from sqlglot.dialects.dialect import create_with_partitions_sql, rename_func
+
+import ibis
+import ibis.common.exceptions as com
+import ibis.expr.operations as ops
+from ibis.backends.base.sqlglot.compiler import NULL, STAR, C, SQLGlotCompiler
+from ibis.backends.base.sqlglot.datatypes import OracleType
+from ibis.backends.base.sqlglot.rewrites import Window, replace_log2, replace_log10
+from ibis.common.patterns import replace
+from ibis.expr.analysis import p, x, y
+from ibis.expr.rewrites import rewrite_sample
+
+
+def _create_sql(self, expression: sge.Create) -> str:
+    # TODO: should we use CREATE PRIVATE instead?  That will set an implicit
+    # lower bound of Oracle 18c
+    properties = expression.args.get("properties")
+    temporary = any(
+        isinstance(prop, sge.TemporaryProperty)
+        for prop in (properties.expressions if properties else [])
+    )
+
+    kind = expression.args["kind"]
+    if (obj := kind.upper()) in ("TABLE", "VIEW") and temporary:
+        if expression.expression:
+            return f"CREATE GLOBAL TEMPORARY {obj} {self.sql(expression, 'this')} AS {self.sql(expression, 'expression')}"
+        else:
+            # TODO: why does autocommit not work here?  need to specify the ON COMMIT part...
+            return f"CREATE GLOBAL TEMPORARY {obj} {self.sql(expression, 'this')} ON COMMIT PRESERVE ROWS"
+
+    return create_with_partitions_sql(self, expression)
+
+
+def _datatype_sql(self: Oracle.Generator, expression: sge.DataType) -> str:
+    # Use this to handle correctly formatting timestamp precision
+    # e.g. TIMESTAMP (scale) WITH TIME ZONE vs. TIMESTAMP WITH TIME ZONE(scale)
+    if expression.is_type("timestamptz"):
+        for exp in expression.expressions:
+            if isinstance(exp, sge.DataTypeParam):
+                return f"TIMESTAMP ({self.sql(exp, 'this')}) WITH TIME ZONE"
+        return "TIMESTAMP WITH TIME ZONE"
+    return self.datatype_sql(expression)
```

Comment:
I don't know if there's a better way to do this, but the regular mapping in `Oracle.Generator.TYPE_MAPPING` didn't expose the scale parameter

Reply:
At some point we should upstream the bug report to sqlglot

Reply:
https://github.com/tobymao/sqlglot/issues/2879